Dessert - Failed edge case

amirahaile opened the pull request:
Took a while to get items to not repeat for Dinner. Tried to create a randomizer method, but found `Array#sample` and learned its limitations (hence the duplication of item arrays). I tried to account for user input edge cases (i.e. non-alpha menu items) but couldn't get an accurate return. I tried a few things that I thought should work, but I don't understand RegEx at all so it was pretty much stabs in the dark.

amirahaile committed:
Breakfast

amirahaile committed:
Lunch

amirahaile committed:
Dinner (1/2): No items are ever repeated.

amirahaile committed:
Dinner (2/2): User decides how many menu items to show.

amirahaile committed:
Dessert

amirahaile committed:
Failed edge case attempt: Deleting non-alpha category items

kariabancroft commented:
I love that you used incremental commits in Git for each level - very nice!

kariabancroft commented on random_menu_generator.rb:
```diff
@@ -0,0 +1,49 @@
+
+# Failed edge case testing.
+#
+# [adjectives, cooking_method, food].each do |array|
```
I see what you're trying to do here. Using the alpha regex matcher was definitely the right idea! Regular expressions are used to attempt to match against a given string or input, so the `match` method is needed to determine whether your word is indeed all alphanumeric characters. I attempted to complete this code in the way that we'd need to match that word below:

``` ruby
array.each do |word|
    unless word.match(/[[:alpha:]]/)
      puts "this one was deleted: #{word}"
      array.delete(word)
    end
```

amirahaile replied:
Ah! So close! I think I kept getting something like a `MatchError` so this is all making sense now. Hopefully I can use it successfully soon. 😊

kariabancroft commented on random_menu_generator.rb:
```diff
@@ -0,0 +1,49 @@
+
+if num_of_items > min_items
+  puts "\nI'm sorry, we're only prepared to generate #{min_items} menu item(s)."
+  num_of_items = min_items
```
This is great! It's definitely important to normalize the lengths when you're letting a user input the data

kariabancroft commented on random_menu_generator.rb:
```diff
@@ -0,0 +1,49 @@
+# ASSEMBLES THE LIST
+num_of_items.times do
+  # don't forget to count by indices (hence [count - 1])
+  puts "#{count}. #{rando_adjective[count - 1]} #{rando_cooking_method[count - 1]} #{rando_food[count - 1]}"
```
If you started your count at 0 on line 43, then you would only need to do `count + 1` on this line once rather than `count - 1` three different times.

amirahaile replied:
hahaha, duh! Should've smelled the opportunity to DRY this one up. 

… that was a weird sentence.

kariabancroft commented:
Good job! I really appreciate your attempt to validate the input via a regex. We will definitely cover this more in the future but hopefully this little code snippet will help you in the immediate.

amirahaile committed:
Revised according to Kari's suggestion

Set `count = 0` on line 43 allowing me to mutate its value only once on line 47.

amirahaile committed:
Fixed regex matcher.

amirahaile committed:
Merge branch 'adh/master' of github.com:amirahaile/C3Projects--RandomMenuGenerator into adh/master

amirahaile commented on random_menu_generator.rb:
```diff
@@ -0,0 +1,47 @@
+# thanks to Kari - figured out RegEx
+# testing that the user inputted only words and not numbers/symbols/etc.
+def validate_alpha_chars(array)
+  array.each do |word|
```
I inverted the regex matcher because it was matching any input that included an alpha character (e.g. "9483c" would pass) and turned it into a method so I could check every instance of user input as soon as it was entered.
